Cabal-syntax/src/Distribution/Types/Flag.hs

```diff
@@ -261,6 +261,17 @@ instance Pretty FlagAssignment where
 -- >>> simpleParsec "+foo -bar baz" :: Maybe FlagAssignment
 -- Nothing
 --
+-- Issue #7279 was fixed in Cabal-3.8
+--
+-- >>> explicitEitherParsec (parsecCommaList parsec) "+foo , -bar" :: Either String [FlagAssignment]
+-- Right [fromList [(FlagName "foo",(1,True))],fromList [(FlagName "bar",(1,False))]]
+--
+-- >>> explicitEitherParsec (parsecCommaList parsecFlagAssignmentNonEmpty) "+foo , -bar" :: Either String [FlagAssignment]
+-- Right [fromList [(FlagName "foo",(1,True))],fromList [(FlagName "bar",(1,False))]]
+--
+-- >>> simpleParsec "+foo+foo" :: Maybe FlagAssignment
```

Comment:
@Mikolaj I hope that `doctest`s are still run as part of CI. i.e. there are tests.

Reply:
Oh, good, So that confirms what I hypothesised in #8040.

Reply:
it does https://github.com/haskell/cabal/runs/5499562852?check_suite_focus=true

afaiu tests will prove this change does not break anything else, but they dont ensure we can break again flags + trailing whitespace

Reply:
oh sorry doctests are inlined and 

`>>> explicitEitherParsec (parsecCommaList parsecFlagAssignmentNonEmpty) "+foo , -bar" :: Either String [FlagAssignment]
-- Right [fromList [(FlagName "foo",(1,True))],fromList [(FlagName "bar",(1,False))]]`

is testing this